internal/advertisement-operator/controller.go

```diff
@@ -93,21 +113,30 @@ func (r *AdvertisementReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
 // check if the advertisement is interesting and set its status accordingly
-func checkAdvertisement(r *AdvertisementReconciler, ctx context.Context, log logr.Logger,
-	adv *protocolv1.Advertisement) {
-
-	//TODO: implement logic
-	adv.Status.AdvertisementStatus = "ACCEPTED"
-	metav1.SetMetaDataAnnotation(&adv.ObjectMeta, "advertisementStatus", "accepted")
+func (r *AdvertisementReconciler) checkAdvertisement(ctx context.Context, log logr.Logger, adv *protocolv1.Advertisement) {
```

Comment:
Due to the importance of this logic, I think that  a dedicated unit Test is needed.